packages/drivers/odsp-driver/src/retryUtils.ts

```diff
@@ -43,8 +43,8 @@ export async function runWithRetry<T>(
 
             const coherencyError = error?.[Odsp409Error] === true;
             const serviceReadonlyError = error?.errorType === OdspErrorType.serviceReadOnly;
-            // Retry for 409 coherency errors or serviceReadOnly errors.
-            if (!(coherencyError || serviceReadonlyError)) {
+            // Retry for retriable 409 coherency errors or serviceReadOnly errors.
+            if (!(coherencyError || serviceReadonlyError || canRetry)) {
```

Comment:
@jatgarg, I do not think this is correct change.
Glancing through the code, these error codes are always retryable, which means someone patches error object with .canRetry = false. There are only two places in code where it happens - this function, and fetchSnapshot(). So it has to be the later.
Note that comment in fetchSnapshot() points out that retrying getVersion() will result in pretty bad results, given that getVersions() has this.firstVersionCall check and will fall back to very expensive blob-by-blob file boot experience on retry.

So I think the right solution here - if canRetry === false, we should throw error - it does not matter what error code we deal with. Assert was catching wrong-doing by this function.